server/aesdsocket.c

```diff
@@ -0,0 +1,342 @@
+    close(socket_fd);
+    closelog();
+    unlink("/var/tmp/aesdsocketdata");
+    exit(0);
+  }
+  
+  signal_received = true;
+}
+
+
+
+
+
+int main(int argc, char *argv[])
+{
+
+  if(argc >= 2)
+    if(strcmp(argv[1],"-d")  == 0)
+      deamon = true;
+
+
+
+  char s[INET6_ADDRSTRLEN];
+  struct addrinfo hints, *conn_info;
+  memset(&hints, 0, sizeof hints);
+  int  ret;
+
+  read_buf = malloc(1024);
+
+  hints.ai_family = AF_INET;
+  hints.ai_socktype = SOCK_STREAM;
+  hints.ai_flags = AI_PASSIVE;
+
+  getaddrinfo(NULL, "9000", &hints, &conn_info);
+
+  socket_fd = socket(conn_info->ai_family,conn_info->ai_socktype, conn_info->ai_protocol);
+  if(socket_fd == -1)
+    return -1;
+
+  ret = bind(socket_fd, conn_info->ai_addr, conn_info->ai_addrlen);
+  if(ret < 0)
+  {
+    perror("bind");
+    return -1;
+  }
+
+  freeaddrinfo(conn_info);
+
+  //Starting the deamon
+  if(deamon)
+    {
+      pid_t pid;
+
+      pid = fork ();
+      if (pid == -1)
+      {
+          free(read_buf);
+          close(socket_fd);
+          perror("fork");
+          return -1;
+      }
+      else if (pid != 0)
+      {
+          exit (EXIT_SUCCESS);
+      }
+
+      else //Valid pid
+      {
+        if(setsid()==-1)
+        {
+            free(read_buf);
+            perror("session");
+            close(socket_fd);
+            return -1;
+        }
+        if(chdir("/")==-1)
+        {
+            free(read_buf);
+            perror("chdir");
+            close(socket_fd);
+            return -1;
+        }
+        close(STDIN_FILENO);
+        close(STDOUT_FILENO);
+        close(STDERR_FILENO);
+        open ("/dev/null", O_RDWR);
+        dup (0); 
+        dup (0); 
+      }
+    }
+
+
+  //Listening on the port
+  ret = listen(socket_fd, 5);  
+  if(ret < 0)
+  {
+    perror("listen");
+    return -1;
+  }
+
+  struct sockaddr_storage their_addr;
+  socklen_t addr_size = sizeof their_addr;
+
+
+  int  new_fd = 0;
+  f_fd = open("/var/tmp/aesdsocketdata",O_RDWR | O_CREAT | O_TRUNC | O_APPEND, S_IRUSR | S_IWUSR | S_IRGRP |S_IWGRP | S_IROTH | S_IWOTH); //666 equivalent value
+
+
+  
+
+  size_t len, total_len;
+  int r_ret = 0;
+  bool packet_complete = false;
+  uint64_t realloc_count = 1;
+
+
+  if (signal (SIGINT, signal_handler) == SIG_ERR) 
+  {
+    perror("signal()");
+    return -1;
+  }
+  if (signal (SIGTERM, signal_handler) == SIG_ERR) 
+  {
+    perror("signal()");
+    return -1;
+  }
+
+
+
+  while(!signal_received)
+  {
+    new_fd = accept(socket_fd, (struct sockaddr *)&their_addr, &addr_size);
+    if(new_fd< 0)
+    {
+      perror("accept");
+      return -1;
+    }
+
+    inet_ntop(their_addr.ss_family,get_in_addr((struct sockaddr *)&their_addr),s, sizeof s);
+    syslog(LOG_DEBUG, "Accepted connection from %s",s);
+    printf("Accepted connection from %s",s);
+
+    finished_writing = true;
+
+    while((!packet_complete))
+    {
+      
+      r_ret = recv(new_fd, (read_buf+((realloc_count-1)*1024)), 1024, 0);
```

Comment:
@kevintom98 you should avoid repeating 1024 everywhere and make this a preprocessor define or constant.

Also I think this assumes that recv will not send partial small packets.  If you do you will put the first part of the packet at the beginning of the read buffer and the next one 1024 bytes into the buffer, with a gap of uninitialized bytes in between.